plugins/modules/pfsense_dns_resolver.py

```diff
@@ -0,0 +1,573 @@
+    # TODO: Strict Outgoing Network interface Binding: check box option
+    system_domain_local_zone_type=dict(default='transparent', choices=['deny', 'refuse', 'static', 'transparent', 'typetransparent', 'redirect', 'inform',
+                                                                       'inform_deny', 'nodefault']),
+    dnssec=dict(default=False, type='bool'),
+    # TODO: Python Module: Enable the Python Module. These 3 options omited when disabled
+    # python=dict(default=False, type='bool'),
+    # python_order=dict(default="pre_validator", type='str', choices=["pre_validator", "post_validator"]),
+    # python_script=dict(default="", type='str'), #Not sure what this is or how to handle it.
+    forwarding=dict(default=False, type='bool'),
+    forward_tls_upstream=dict(default=False, type='bool'),
+    regdhcp=dict(default=False, type='bool'),
+    regdhcpstatic=dict(default=False, type='bool'),
+    regovpnclients=dict(default=False, type='bool'),
+    custom_options=dict(default="", type='str'),
+    hosts=dict(default=[], type='list', elements='dict', options=DNS_HOST_SPEC),
+    domainoverrides=dict(default=[], type='list', elements='dict', options=DNS_DOMAIN_OVERRIDE_SPEC),
+    # Advanced Settings
+    hideidentity=dict(default=True, type='bool'),
+    hideversion=dict(default=True, type='bool'),
+    # TODO: Query Name Minimization
+    # TODO: Strict Query Name Minimization
+    prefetch=dict(default=False, type='bool'),
+    prefetchkey=dict(default=False, type='bool'),
+    dnssecstripped=dict(default=True, type='bool'),
+    # TODO: Serve Expired
+    # TODO: Aggressive NSEC
+    msgcachesize=dict(default=4, type='int', choices=[4, 10, 20, 50, 100, 250, 512]),
+    outgoing_num_tcp=dict(default=10, type='int', choices=[0, 10, 20, 30, 50]),
+    incoming_num_tcp=dict(default=10, type='int', choices=[0, 10, 20, 30, 50]),
+    edns_buffer_size=dict(default="4096", type='str', choices=["auto", "512", "1220", "1232", "1432", "1480", "4096"]),
+    num_queries_per_thread=dict(default=512, type='int', choices=[512, 1024, 2048]),
+    jostle_timeout=dict(default=200, type='int', choices=[100, 200, 500, 1000]),
+    cache_max_ttl=dict(default=86400, type='int'),
+    cache_min_ttl=dict(default=0, type='int'),
+    infra_host_ttl=dict(default=900, type='int', choices=[60, 120, 300, 600, 900]),
+    infra_cache_numhosts=dict(default=10000, type='int', choices=[1000, 5000, 10000, 20000, 50000, 100000, 200000]),
+    unwanted_reply_threshold=dict(default="disabled", type='str', choices=["disabled", "5000000", "10000000", "20000000", "40000000", "50000000"]),
+    log_verbosity=dict(default=1, type='int', choices=[0, 1, 2, 3, 4, 5])
+    # TODO: Disable Auto-added Access Control
+    # TODO: Disable Auto-added Host Entries
+    # TODO: Experimental Bit 0x20 Support
+    # TODO: DNS64 Support
+)
+
+dns_REQUIRED_IF = []
+
+
+class PFSenseDNSModule(PFSenseModuleBase):
```

Comment:
I'm wondering if this shouldn't be DNSResolverModule and DNS_RESOLVER_ for the vars.

Reply:
This makes sense. I changed the naming in this file.